identity-appsupport/src/commonMain/kotlin/com/android/identity/appsupport/ui/consent/ConsentModalBottomSheet.kt

```diff
@@ -0,0 +1,465 @@
+                )
+            }
+            if (relyingParty.trustPoint.displayName != null) {
+                Text(
+                    text = relyingParty.trustPoint.displayName!!,
+                    style = MaterialTheme.typography.titleLarge,
+                    fontWeight = FontWeight.Bold,
+                )
+            }
+        } else if (relyingParty.websiteOrigin != null) {
+            Text(
+                text = stringResource(
+                    Res.string.consent_modal_bottom_sheet_headline_share_with_known_requester,
+                    relyingParty.websiteOrigin
+                ),
+                style = MaterialTheme.typography.titleLarge,
+                fontWeight = FontWeight.Bold,
+            )
+        } else {
+            Text(
+                text = stringResource(Res.string.consent_modal_bottom_sheet_headline_share_with_unknown_requester),
+                style = MaterialTheme.typography.titleLarge,
+                fontWeight = FontWeight.Bold,
+            )
+        }
+        Spacer(modifier = Modifier.height(16.dp))
+    }
+}
+
+@OptIn(ExperimentalResourceApi::class)
+@Composable
+private fun DocumentSection(document: ConsentDocument) {
+    Column(
+        modifier = Modifier
+            .padding(vertical = 2.dp)
+            .fillMaxWidth(),
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .clip(shape = RoundedCornerShape(16.dp, 16.dp, 0.dp, 0.dp))
+                .background(MaterialTheme.colorScheme.surfaceContainerLowest),
+            horizontalAlignment = Alignment.Start,
+        ) {
+            Row(
+                modifier = Modifier.padding(16.dp)
+            ) {
+                Icon(
+                    modifier = Modifier.size(50.dp),
+                    bitmap = document.cardArt.decodeToImageBitmap(),
+                    contentDescription = stringResource(Res.string.consent_modal_bottom_sheet_card_art_description),
+                    tint = Color.Unspecified
+                )
+                Column(
+                    modifier = Modifier.padding(start = 16.dp)
+                ) {
+                    Text(
+                        text = document.name,
+                        style = MaterialTheme.typography.bodyMedium,
+                        fontWeight = FontWeight.Bold
+                    )
+                    Text(
+                        text = document.description,
+                        style = MaterialTheme.typography.bodySmall,
+                        color = MaterialTheme.colorScheme.secondary
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalTextApi::class)
+@Composable
+private fun RequestSection(
+    consentFields: List<ConsentField>,
+    relyingParty: ConsentRelyingParty
+) {
+    val dontUseColumns = consentFields.size <= 5
+    val storedFields = consentFields.filter { it is MdocConsentField && it.intentToRetain == true }
```

Comment:
Maybe to make it clear that all elements are included in these two results, use partition() instead of filter()?

val (storedFields, notStoredFields) = consentFields.partition { it is MdocConsentField && it.intentToRetain == true }

...then there's also no risk that a slight change or bug in either condition could end up excluding some items.

Reply:
Nice - TIL!